### DISCUSSION TITLE
Question on: PolarFire SoC: Programming the SPI Flash on the Icicle Kit and Booting via HSS

### DISCUSSION THREAD

**Discussion by andrenemat** [polarfire-soc/polarfire-soc-documentation#106] [HSS and U-Boot] [answered]
Does the PolarFire SoC: Programming the SPI Flash on the Icicle Kit and Booting via HSS example run on an Icicle kit with ES silicon. The reason for my question is the errata item in the ES silicon that the MSS cannot access the System Controller SPI.

Thanks a lot

**Comment by hughbreslin** [comment, marked as answer]
Hey @andrenemat 

From testing ourselves, you can boot a HSS payload programmed into the system controller SPI flash following the steps in the [Programming the SPI Flash on the Icicle Kit and Booting via HSS](https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/boards/mpfs-icicle-kit-es/programming-spi-flash-via-libero/programming-spi-flash-via-libero.md) document. Its our understanding that you cannot write to the system controller SPI flash via the MSS. 

This also the flow demonstrated in the [bitstream builder](https://github.com/polarfire-soc/icicle-kit-minimal-bring-up-design-bitstream-builder) demo.

Its not clear though looking at section 3.1 of the [errata doc](https://www.microsemi.com/document-portal/doc_download/1245049-er0219-polarfire-soc-fpga-engineering-samples-es-devices). Let me double check and get clarification for you to make sure that our understanding is correct (we haven't seen any issues so far).

**Reply by andrenemat** [reply]
Thanks a lot @hughbreslin

**Reply by hughbreslin** [reply]
Just for clarification after double checking, the reads will work, writes will fail. If you do need write access to the SC SPI flash you can access it via the fabric too where both reads and writes will work :) As the errata says this is only an ES issue and won't affect production silicon.

**Reply by andrenemat** [reply]
Thanks a lot @hughbreslin for confirming.
